### PR TITLE
UI for submission navigation

### PR DESCRIPTION
**Pull request by nzcodarnoc**
Big shoutout to @OBaddeley for helping me with this.

Works as you'd expect.

There's some validation on the text field, error handling is a bit rough for now.

image

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -122,6 +122,22 @@ def survey_id
...
+  def next
+    consultation.submissions.find_by("id > ?", id)
+  end
+
+  def prev
+    consultation.submissions.find_by("id < ?", id)
+  end
+
+  def first
+    consultation.submissions.first
```
Comment (mermop) [inline app/models/submission.rb:134]: These are vulnerable to the default scope changing the ordering - I'd make it explicit. Also, we probably only want active (not archived) submissions

```suggestion
    consultation.submissions.active.order(:created_at).first
```

File: app/models/submission.rb [hunk 2]
```diff
@@ -122,6 +122,22 @@ def survey_id
...
+  end
+
+  def last
+    consultation.submissions.last
```
Comment (mermop) [inline app/models/submission.rb:138]: as above - making the order explicit and only using active submissions

```suggestion
    consultation.submissions.active.order(:created_at).last
```

File: app/views/submissions/_submission_navigation.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<form method="get" action="goto">
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:1]: This looks great, nice work

File: app/views/submissions/_submission_navigation.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  <div class="input-group">
+    <div class="input-group-prepend">
+      <a title="Go to first submission" class="btn <%= submission.prev ? "" : "disabled" %>" href="<%= consultation_submission_tag_path(submission.first.consultation, submission.first) %>">
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:4]: The consultation of the first and last submission will always be the same as the current one

```suggestion
      <a title="Go to first submission" class="btn <%= submission.prev ? "" : "disabled" %>" href="<%= consultation_submission_tag_path(submission.consultation, submission.first) %>">
```

File: app/views/submissions/_submission_navigation.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        <svg id="i-start" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" width="20" height="20" fill="none" stroke="black" stroke-linecap="round" stroke-linejoin="round" stroke-width="3">
+          <path d="M8 2 L8 16 22 2 22 30 8 16 8 30" />
+        </svg>
+      </a>
+      <a title="Go to previous submission" class="btn <%= submission.prev ? "" : "disabled" %>" href="<%= submission.prev ? consultation_submission_tag_path(submission.prev.consultation, submission.prev) : "#" %>">
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:9]:
```suggestion
      <a title="Go to previous submission" class="btn <%= submission.prev ? "" : "disabled" %>" href="<%= submission.prev ? consultation_submission_tag_path(submission.consultation, submission.prev) : "#" %>">
```

File: app/views/submissions/_submission_navigation.erb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <svg id="i-caret-left" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" width="20" height="20" fill="none" stroke="black" stroke-linecap="round" stroke-linejoin="round" stroke-width="3">
+          <path d="M22 30 L6 16 22 2 Z" />
+        </svg>
+      </a>
+      <label class="input-group-text" for="inputSubmissionId">ID</label>
+    </div>
+    <input name="goto_submission_id" type="number" min="<%= submission.first.id %>" max="<%= submission.last.id %>" value="<%= submission.id %>" class="form-control" id="inputSubmissionId">
+    <input name="goto_consultation_id" type="hidden" value="<%= submission.consultation.id %>">
+    <div class="input-group-append">
+      <a title="Go to next submission" class="btn <%= submission.next ? "" : "disabled" %>" href="<%= submission.next ? consultation_submission_tag_path(submission.next.consultation, submission.next) : "#" %>">
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:19]:
```suggestion
      <a title="Go to next submission" class="btn <%= submission.next ? "" : "disabled" %>" href="<%= submission.next ? consultation_submission_tag_path(submission.consultation, submission.next) : "#" %>">
```

File: app/views/submissions/_submission_navigation.erb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        <svg id="i-caret-right" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" width="20" height="20" fill="none" stroke="black" stroke-linecap="round" stroke-linejoin="round" stroke-width="3">
+          <path d="M10 30 L26 16 10 2 Z" />
+        </svg>
+      </a>
+      <a title="Go to last submission" class="btn <%= submission.next ? "" : "disabled" %> href="<%= consultation_submission_tag_path(submission.last.consultation, submission.last) %>">
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:24]:
```suggestion
      <a title="Go to last submission" class="btn <%= submission.next ? "" : "disabled" %> href="<%= consultation_submission_tag_path(submission.consultation, submission.last) %>">
```

File: app/views/submissions/_submission_navigation.erb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        <svg id="i-end" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" width="20" height="20" fill="none" stroke="black" stroke-linecap="round" stroke-linejoin="round" stroke-width="3">
+          <path d="M24 2 L24 16 10 2 10 30 24 16 24 30" />
+        </svg>
+      </a>
+    </div>
+  </div>
+</form>
```
Comment (mermop) [inline app/views/submissions/_submission_navigation.erb:31]: Unix line ending please

**Comment by mermop** [comment]
This will need some tests, but that'll be a shlep for you Connie - let me know when you've got half an hour, we can pair on them

File: app/views/submissions/_submission_navigation.erb (hunk 8 above)
Comment (nzcodarnoc) [reply app/views/submissions/_submission_navigation.erb]: Done

**Pull request merged**